### PR TITLE
Update promise style to remove redundancy.

### PR DESCRIPTION
**Pull request by lightsofapollo**

### CODE REVIEW

File: models/users.js [hunk 1]
```diff
@@ -20,17 +20,11 @@ var Users = {
...
-          resolve(false);
-        }
-      }).catch(function(error) {
-        reject(error);
-      });
+    return Database.getUser(email).then((result) => {
+      if (!result) {
+        return false;
+      }
+      return new User(result.email, result.password, result.name);
```
Comment (dhylands) [inline models/users.js]: This changes what the function returns, so the documentation block for the function should be updated as well.

Comment (dhylands) [inline models/users.js]: Or I'm just not understanding what's happening here (quite possible).

Comment (lightsofapollo) [inline models/users.js]: Returning from inside of the .then clause is like calling resolve in the previous code. This has the added benefit of resolving promises for you to allow the common promise chain style (which breaks the pyramid of doom).

**Pull request merged**